sql/core/src/test/scala/org/apache/spark/sql/CharVarcharTestSuite.scala

```diff
@@ -926,4 +926,56 @@ class DSV2CharVarcharTestSuite extends CharVarcharTestSuite
       }
     }
   }
+
+  test("SPARK-42611: check char/varchar length in reordered nested structs") {
+    Seq("CHAR(5)", "VARCHAR(5)").foreach { typ =>
+      withTable("t") {
+        sql(s"CREATE TABLE t(s STRUCT<n_c:$typ,n_i:INT>) USING $format")
```

Comment:
extremely nit: shall we have an empty space in between of `n_c:$typ,` and `n_i:INT` in this line and also in the `CREATE TABLE` statements below?

Reply:
Will do, I overlooked the formatting in this class.

Reply:
Done.